File: app/src/main/java/com/example/taskapp/data/ApiService.java
package com.example.taskapp.data;

import java.util.List;
import retrofit2.Call;
import retrofit2.http.*;

public interface ApiService {
    @POST("register")
    Call<ApiResponse> register(@Body AuthRequest request);

    @POST("login")
    Call<ApiResponse> login(@Body AuthRequest request);

    @POST("tasks/list")
    Call<List<Task>> getTasks(@Body AuthRequest request);

    @POST("tasks")
    Call<Task> createTask(@Body TaskCreateRequest request);

    @POST("tasks/{id}/update")
    Call<Task> updateTask(@Path("id") int id, @Body TaskUpdateRequest request);

    @POST("tasks/{id}/delete")
    Call<ApiResponse> deleteTask(@Path("id") int id, @Body TaskDeleteRequest request);
}

File: app/src/main/java/com/example/taskapp/data/ApiService.kt
package com.example.taskapp.data

import retrofit2.Response
import retrofit2.http.*

interface ApiService {
    @POST("register")
    suspend fun register(@Body request: AuthRequest): Response<ApiResponse>

    @POST("login")
    suspend fun login(@Body request: AuthRequest): Response<ApiResponse>

    @POST("tasks/list")
    suspend fun getTasks(@Body request: AuthRequest): Response<List<Task>>

    @POST("tasks")
    suspend fun createTask(@Body request: TaskCreateRequest): Response<Task>

    @POST("tasks/{id}/update")
    suspend fun updateTask(@Path("id") id: Int, @Body request: TaskUpdateRequest): Response<Task>

    @POST("tasks/{id}/delete")
    suspend fun deleteTask(@Path("id") id: Int, @Body request: TaskDeleteRequest): Response<ApiResponse>
}